### ISSUE TITLE
Test hangs till 60 seconds when leave application.

### ISSUE THREAD

**Issue by shynkevich-alex** [facebook/WebDriverAgent#719]
## Environment
* Mac OS 10.12.6 (16G29)
* iOS
* Real device
* PlatformVersion 10.3.3

## Details
Test just clicks on link and safari opens and I am waiting that element (by accessibilityId) can't be found or it is not visible.
I want to continue test but sometimes I can't find elements after it at all.

## Logs
`[Xcode] 2017-09-15 19:49:29.133900+0300 XCTRunner[576:39750] Enqueue Failure: Failure getting snapshot Error Domain=XCTDaemonErrorDomain Code=12 "Failed to get main window after 30 retries: kAXErrorServerNotFound (see <rdar://problem/15530121>)" UserInfo={NSLocalizedDescription=Failed to get main window after 30 retries: kAXErrorServerNotFound (see <rdar://problem/15530121>)} <unknown> 0 1
`

https://gist.github.com/shynkevich-alex/f9a3327801d3fd8c8429f4f3275257c8

**Comment by marekcirkos** [comment]
Yeah, This is frustrating, but what happens is you hit query during animation, which will not behave well and block XCTest for a while. Very tricky to solve. 

All I can advice is to add ugly sleep after tapping link
